apps/frontend-manage/src/pages/courses/[id].tsx

```diff
@@ -21,18 +26,25 @@ function CourseOverviewPage() {
   const theme = useContext(ThemeContext)
 
   const [descriptionEditMode, setDescriptionEditMode] = useState(false)
+  const [isColorPickerVisible, setIsColorPickerVisible] = useState(false)
 
   const { loading, error, data } = useQuery(GetSingleCourseDocument, {
     variables: { courseId: router.query.id as string },
     skip: !router.query.id,
   })
 
+  const [changeCourseColor] = useMutation(ChangeCourseColorDocument)
+
   useEffect(() => {
     if (data && !data.course) {
       router.push('/404')
     }
   }, [data, router])
 
+  const toggleColorPicker = useCallback(() => {
```

Comment:
Is there a specific reason for useCallback here? I think we could just call it like setIsColorPickerVisible(prevState => !prevState)

Reply:
I saw this in multiple similar examples in other projects. As far as I understood, this use of the hook could be seen as a performance optimization, since otherwise the `toggleColorPicker` function would be unnecessarily re-created on every re-render. This way, it will be cached and only updated if there is a change in its dependencies.  

But I also have no issue with leaving it out if you wish :)